Sources/ComposableArchitecture/RootStore.swift

```diff
@@ -64,15 +62,10 @@ public final class RootStore {
           let uuid = UUID()
           let effectCancellable = withEscapedDependencies { continuation in
             publisher
-              .handleEvents(
-                receiveCancel: { [weak self] in
-                  threadCheck(status: .effectCompletion(action))
-                  self?.effectCancellables[uuid] = nil
-                }
-              )
+              .receive(on: UIScheduler.shared)
```

Comment:
While the compiler ensures access on the main thread when using modern Swift concurrency, older APIs like Combine do allow for escape hatches. I've added this `receive(on:)` to this internal Combine code to ensure the store code is really performed on the main actor, and as we de-combine-ify the library in the future we won't have to worry about edge cases like these as much.